docs/sources/clients/promtail/stages/json.md

```diff
@@ -97,6 +97,30 @@ and append the following key-value pairs to the set of extracted data:
 
 - `user`: `marco`
 
+Alternatively you can use a `.` between keys to reference sub-keys in a singe pipeline stage:
+
+```yaml
+- json:
+    expressions:
+      output:    log
+      stream:    stream
+      timestamp: time
+      extra.user:
+```
+so given log line:
+
+```
+{"log":"log message\n","stream":"stderr","time":"2019-04-30T02:12:41.8443515Z","extra":"{\"user\":\"marco\"}"}
+```
+
+This single stage would create the following key-value pairs in the set of
+extracted data:
+
+- `output`: `log message\n`
+- `stream`: `stderr`
+- `timestamp`: `2019-04-30T02:12:41.8443515`
+- `extra.user`: `marco`
```

Comment:
while it extract correctly the data, this example can be confusing since extra.user is not a valid label name :

> Label names may contain ASCII letters, numbers, as well as underscores. They must match the regex [a-zA-Z_][a-zA-Z0-9_]*. Label names beginning with __ are reserved for internal use.

This means the user will need to rename the key with:

```
- json:
    expressions:
      output:    log
      stream:    stream
      timestamp: time
      extra.user:
- labels:
      extra_user:  extra.user
```

The example `JMESPath` cover this too.

Reply:
Hi,
Yes, this is exactly what we do with it. We pull a value from a JSON object that is 4 layers deep and then add it as a label. When reading the current documentation and example, without knowing dot referencing was possible users think that they have to keep adding multiple json stages and referencing the source as the layer above. 
I thought adding an example with dot referencing would help others not make that mistake.

Reply:
I think we are in agreement this is nice to add to the docs, but @cyriltovena is correct, the example as submitted cannot work because the label name cannot have a `.` in it. 

Would you be able to update the example as in Cyril's example in

Reply:
Hi @slim-bean,

I'm confused. This is valid configuration I have in production on hundreds of hosts.

"extra.user" is not a label, it's a json field. The label is "extra_user" which is a valid label and contains the value from the json field.

The whole point of my suggestion was to include json fields with "." in to show users that you can go into a nested json object however deep and pull it up to whatever label you want. I do this on a data structure 4 levels deep myself and the docs weren't clear that it was possible.